### ISSUE TITLE
CMake test failures on CI due to Azure image update

### ISSUE THREAD

**Issue by TheQwertiest** [mesonbuild/meson#7307]
Two cmake module tests are failing on azure agents.

#### `test cases/cmake/5 object library`:
**Failure:** can't find zlib headers:
```
5 object library\subprojects\cmobjlib\libb.cpp(2): fatal error C1083: Cannot open include file: 'zlib.h': No such file or directory
```
**Reason:** `C:\Program Files\PostgreSQL\12\bin` was added to `PATH` on the agent, which contains `zlib.lib`, but not respective headers. It seems PATH has a priority over pkg-config in CMake.
</br>
#### `test cases/cmake/2 advanced`:
**Failure:**
```
 Traceback (most recent call last):
   File "D:\a\1\s\mesonbuild\mesonmain.py", line 131, in run
     return options.run_func(options)
   File "D:\a\1\s\mesonbuild\msetup.py", line 245, in run
     app.generate()
   File "D:\a\1\s\mesonbuild\msetup.py", line 159, in generate
     self._generate(env)
   File "D:\a\1\s\mesonbuild\msetup.py", line 192, in _generate
     intr.run()
   File "D:\a\1\s\mesonbuild\interpreter.py", line 4469, in run
     super().run()
   File "D:\a\1\s\mesonbuild\interpreterbase.py", line 493, in run
     self.evaluate_codeblock(self.ast, start=1)
   File "D:\a\1\s\mesonbuild\interpreterbase.py", line 518, in evaluate_codeblock
     raise e
   File "D:\a\1\s\mesonbuild\interpreterbase.py", line 511, in evaluate_codeblock
     self.evaluate_statement(cur)
   File "D:\a\1\s\mesonbuild\interpreterbase.py", line 526, in evaluate_statement
     self.assignment(cur)
   File "D:\a\1\s\mesonbuild\interpreterbase.py", line 1179, in assignment
     value = self.evaluate_statement(node.value)
   File "D:\a\1\s\mesonbuild\interpreterbase.py", line 528, in evaluate_statement
     return self.method_call(cur)
   File "D:\a\1\s\mesonbuild\interpreterbase.py", line 923, in method_call
     return obj.method_call(method_name, args, self.kwargs_string_keys(kwargs))
   File "D:\a\1\s\mesonbuild\interpreter.py", line 1813, in method_call
     value = fn(self.interpreter, state, args, kwargs)
   File "D:\a\1\s\mesonbuild\interpreterbase.py", line 286, in wrapped
     return f(*wrapped_args, **wrapped_kwargs)
   File "D:\a\1\s\mesonbuild\interpreterbase.py", line 213, in wrapped
     return f(*wrapped_args, **wrapped_kwargs)
   File "D:\a\1\s\mesonbuild\interpreterbase.py", line 182, in wrapped
     return f(*wrapped_args, **wrapped_kwargs)
   File "D:\a\1\s\mesonbuild\modules\cmake.py", line 296, in subproject
     subp = interpreter.do_subproject(dirname, 'cmake', kwargs)
   File "D:\a\1\s\mesonbuild\interpreter.py", line 2823, in do_subproject
     raise e
   File "D:\a\1\s\mesonbuild\interpreter.py", line 2809, in do_subproject
     return self._do_subproject_cmake(dirname, subdir, subdir_abs, default_options, kwargs)
   File "D:\a\1\s\mesonbuild\interpreter.py", line 2870, in _do_subproject_cmake
     cm_int.analyse()
   File "D:\a\1\s\mesonbuild\cmake\interpreter.py", line 976, in analyse
     i.postprocess(self.output_target_map, self.src_dir, self.subdir, self.install_prefix, self.trace)
   File "D:\a\1\s\mesonbuild\cmake\interpreter.py", line 466, in postprocess
     self.sys_includes = list(OrderedSet([rel_path(x, True, False) for x in OrderedSet(self.sys_includes)]))
   File "D:\a\1\s\mesonbuild\cmake\interpreter.py", line 466, in <listcomp>
     self.sys_includes = list(OrderedSet([rel_path(x, True, False) for x in OrderedSet(self.sys_includes)]))
   File "D:\a\1\s\mesonbuild\cmake\interpreter.py", line 437, in rel_path
     and os.path.commonpath([x, self.env.get_source_dir()]) == self.env.get_source_dir()
   File "C:\hostedtoolcache\windows\Python\3.5.4\x64\lib\ntpath.py", line 632, in commonpath
     raise ValueError("Paths don't have the same drive")
 ValueError: Paths don't have the same drive
```

**Reason:** not sure, but most likely caused by using `rel_path` on `sys_includes` and `source_dir`, where `sys_includes` has a zlib from Postgre (which is on disk `C:`, while `source_dir` is on disk `D:`)

CC @mensinda

**Comment by nirbheek** [comment]
`2 advanced` is a good breakage, it tells us something useful. `5 object library` is just the image changing underneath us, afaict.

**Comment by TheQwertiest** [comment]
@nirbheek , I would say `5` is an issue as well: user should be able to specify the source of the dependency (e.g. not just a random system library, but the one that *user* wants). It's a huge thing, IMO.

**Comment by mensinda** [comment]
I also have to investigate why `2 advanced` fails during configure and why `5 object library` only fails while building (missing include paths, that should theoretically be there).

**Comment by TheQwertiest** [comment]
> missing include paths, that should theoretically be there
I think this is actually a proper behavior: build system can't know which header is needed unless it was stated explicitly. In this case CMake(?) found some dir with the name `include`, which is in the same place as the relevant `lib` dir, but it didn't know that `zlib.h` should've been there.

**Comment by mensinda** [comment]
The include dir *is* set in CMake: `'INTERFACE_INCLUDE_DIRECTORIES': ['C:/Program Files/PostgreSQL/12/include']`

So meson should also know about it.

However, I would prefer to skip the `2 advanced` bug for now, since I like to fix this by migrating to pathlib, which will take a bit more time.

**Comment by TheQwertiest** [comment]
> So meson should also know about it.

I think the problem is not that meson doesn't know about the dir, but rather that that dir does not contain `zlib.h` (I mean it's Postgre not a zlib =) ).

**Comment by mensinda** [comment]
> [EDIT]: Yep, dir is included in search paths: `-IC:/Program$ Files/PostgreSQL/12/include`

I guess I have one less bug to worry about :)

Windows is just the best OS for everything.

**Comment by TheQwertiest** [comment]
> Windows is just the best OS for everything.

I so love starting holy-wars, but I will refrain (... for now :P)

**Comment by mensinda** [comment]
Fixed with #7316. See #7319 for the followup PR.
